src/measure_theory/integral/circle_integral_transform.lean

```diff
@@ -0,0 +1,173 @@
+is holomorphic.
+
+-/
+
+open set measure_theory metric filter function
+open_locale interval real
+
+noncomputable theory
+
+variables {E : Type} [normed_group E] [normed_space ℂ E] (R : ℝ) (z w : ℂ)
+
+namespace complex
+
+/-- Given a function `f : ℂ → E`, this gives the function $(2πi)^{-1}\frac{f(x)}{x-w}$ where `x`
+runs over a circle of radius `R` around `z`. If `f` is differentiable and `w` is in the interior of
+the ball, then the integral from `0` to `2 * π` of this gives the value `f(w)`. -/
+def circle_integral_transform (f : ℂ → E) : ℝ → E :=
+  λ θ, (2 * ↑π * I)⁻¹ • deriv (circle_map z R) θ • ((circle_map z R θ) - w)⁻¹ • f (circle_map z R θ)
+
+/-- The derivative of `circle_integral_transform` w.r.t `w` -/
+def circle_integral_transform_deriv (f : ℂ → E) (θ : ℝ) : E :=
+  (2 * ↑π * I)⁻¹ • deriv (circle_map z R) θ • ((circle_map z R θ - w) ^ 2)⁻¹ • f (circle_map z R θ)
+
+lemma circle_integral_transform_deriv_periodic (f : ℂ → E) :
+  periodic (circle_integral_transform_deriv R z w f) (2 * π) :=
+begin
+  have := periodic_circle_map,
+  simp_rw periodic at *,
+  intro x,
+  simp_rw [circle_integral_transform_deriv, this],
+  congr' 2,
+  simp [this],
+end
+
+lemma circle_integral_transform_deriv_eq (f : ℂ → E) :
+  circle_integral_transform_deriv R z w f =
+  (λ θ, (circle_map z R θ - w)⁻¹ • (circle_integral_transform R z w f θ)) :=
+begin
+  ext,
+  simp_rw [circle_integral_transform_deriv, circle_integral_transform, ←mul_smul, ←mul_assoc],
+  ring_nf, simp,
+end
+
+lemma circle_integral_transform_circle_int [complete_space E] (f : ℂ → E) :
+  ∫ (θ : ℝ) in 0..2 * π, circle_integral_transform R z w f θ =
+  (2 * ↑π * I)⁻¹ • ∮ z in C(z, R), (z - w)⁻¹ • f z :=
+begin
+  simp_rw [circle_integral_transform, circle_integral, deriv_circle_map, circle_map],
+  simp,
+end
+
+lemma circle_integral_transform_continuous {R : ℝ} (hR : 0 < R) {f : ℂ → E} {z w : ℂ}
+  (hf : continuous_on f $ sphere z R) (hw : w ∈ ball z R) :
+  continuous (circle_integral_transform R z w f) :=
+begin
+  rw circle_integral_transform,
+  apply_rules [continuous.smul, continuous_const],
+  simp_rw deriv_circle_map,
+  apply_rules [continuous.mul, (continuous_circle_map 0 R), continuous_const],
+  { apply circle_map_inv_continuous hw },
+  { apply continuous_on.comp_continuous hf (continuous_circle_map z R),
+    exact (λ _, (circle_map_mem_sphere _ hR.le) _) },
+end
+
+lemma circle_integral_transform_deriv_continuous {R : ℝ} (hR : 0 < R) {f : ℂ → E} {z w : ℂ}
+  (hf : continuous_on f (sphere z R)) (hw : w ∈ ball z R) :
+  continuous (circle_integral_transform_deriv R z w f) :=
+begin
+  rw circle_integral_transform_deriv_eq,
+  exact (circle_map_inv_continuous hw).smul (circle_integral_transform_continuous hR hf hw),
+end
+
+/--A useful bound for circle integrals (with complex codomain)-/
+def circle_integral_bounding_function (R : ℝ) (z : ℂ) : (ℂ × ℝ → ℂ) :=
+  λ w, circle_integral_transform_deriv R z w.1 (λ x, 1) w.2
```

Comment:
```suggestion
def circle_integral_bounding_function (R : ℝ) (z : ℂ) (w : ℂ × ℝ) : ℂ :=
circle_integral_transform_deriv R z w.1 (λ x, 1) w.2
```
Often, it is just more convenient to define a function by putting all variables before the colon.